turn handler, add logic to game.rb and game_spec.rb

PamBWillenz opened the pull request:
Add turn handler logic 
game.rb, game_spec.rb

will add directions in games controller

can add directions in pieces controller, too (is this necessary?)

lasley commented on app/models/game.rb:
```diff
@@ -49,4 +49,16 @@ def get_piece(x, y)
     self.pieces.where(x_position: x, y_position: y).take
   end
 
+  def set_default_turn
```
This should probably be a bang method.
https://stackoverflow.com/questions/612189/why-are-exclamation-marks-used-in-ruby-methods

kellylougheed commented on spec/models/game_spec.rb:
```diff
@@ -36,6 +36,19 @@
+    let(:piece) { FactoryGirl.create(:piece, game_id: game.id, player_id: white_player.id) }
+
+    it "should change turns after player moves" do
+    expect(game.set_default_turn).to eq(true)
```
Does this "true" just mean a truthy value? Otherwise, we'd expect it to be the white player id, right?

PamBWillenz replied:
Yes, @kellylougheed - you're absolutely right - I tested it with byebug and it should equal white player id. I'll change it and hope it does equal the white player instead of just being true.

kellylougheed commented:
Pam, this is looking good! For tests, maybe break it down into a couple specific tests? You could use different tests to make sure that set_default_turn works, that current_turn changes after a move, and that opponent shows the opponent correctly. I like how you broke down all the methods in game.rb!

PamBWillenz commented:
Yep - I can do that, Kelly. I kind of combine all three of the methods into that one test.

PamBWillenz commented:
Added more tests to cover all the conditions of a player's turn

- I had to change `current_turn` method to `current_player_turn` because I was getting a stack too deep error with the first name. I think because it matches our datatype in game schema, it was confused.

- I think `current_player_turn` does the same logic that the `opponent` method is doing. I added it to help Matt connect some JS stuff. 

- My tests are failing because I am either getting false for both colors when it is someone's turn or the wrong id is coming up. I used a factory to give id's because every reload would change the id's

- I need to add logic to piece.rb and test and also add code in the game#controller#update to connect the `current_turn` to a player_id.

PamBWillenz commented:
Got all tests in game#spec and piece#spec passing. Add a line of logic in game#controller#update to set the default turn after a game is created and also put a line of logic in the `move_to` in piece.rb (wrote a test for that one).

joewhale commented:
Looks good. Works well with JS!
